pkg/cli/debug/debug.go

```diff
@@ -336,13 +336,13 @@ func (o *DebugOptions) RunDebug() error {
 	if len(o.FilenameOptions.Filenames) > 0 {
 		b.FilenameParam(o.ExplicitNamespace, &o.FilenameOptions)
 	}
-	one := false
-	infos, err := b.Do().IntoSingleItemImplied(&one).Infos()
+	infos, err := b.Do().Infos()
 	if err != nil {
 		return err
 	}
-	if !one {
-		return fmt.Errorf("you must identify a resource with a pod template to debug")
+	if len(infos) != 1 {
+		klog.V(4).Infof("Objects: %#v", infos)
```

Comment:
In general singletons that work on a single object we don't name the total (because it could be hundreds).  I had the debugging line because I didn't trust infos when I was piping input (that's what triggered the bug this fixes, because stdin is not SingleItemImplied, so you couldn't pipe a single pod to this).